src/trans/gpu/internal/tpm_stats.F90

```diff
@@ -0,0 +1,61 @@
+! (C) Copyright 2000- ECMWF.
+! (C) Copyright 2022- NVIDIA.
+! 
+! This software is licensed under the terms of the Apache Licence Version 2.0
+! which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+! In applying this licence, ECMWF does not waive the privileges and immunities
+! granted to it by virtue of its status as an intergovernmental organisation
+! nor does it submit to any jurisdiction.
+!
+
+MODULE TPM_STATS
+
+IMPLICIT NONE
+
+CHARACTER(LEN=32) :: DESCRIPTIONS(100)
+
+CONTAINS
+
+SUBROUTINE GSTATS_LABEL_NVTX(KNUM,CTYPE,CDESC)
+USE EC_PARKIND  ,ONLY : JPIM
+IMPLICIT NONE
+INTEGER(KIND=JPIM) :: KNUM
+CHARACTER(*) CDESC
+CHARACTER(*) CTYPE
+
+IF (KNUM >= 400 .AND. KNUM < 500) THEN
+  DESCRIPTIONS(KNUM-400+1) = CDESC
+ENDIF
+CALL GSTATS_LABEL(KNUM,CTYPE,CDESC)
+END SUBROUTINE
+
+SUBROUTINE GSTATS_NVTX(KNUM,KSWITCH)
```

Comment:
Not sure you actually want this. This is a wrapper around the normal gstats to enable nvtx support (nsight system)